### PR TITLE
Add State Batch Delete for DB

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds functions to batch delete in DB. This will be used for pruning beacon state stored in DB prior to finalized check point

### CODE REVIEW

File: beacon-chain/db/kv/state_test.go [hunk 1]
```diff
@@ -113,3 +115,50 @@ func TestGenesisState_CanSaveRetrieve(t *testing.T) {
...
+	for _, r := range blockRoots {
+		s, err := db.State(context.Background(), r)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if s == nil {
+			continue
+		}
+		if s.Slot%2 == 0 {
+			t.Errorf("State with slot %d should have been deleted", s.Slot)
```
Comment (shayzluf) [inline beacon-chain/db/kv/state_test.go]: nice test

File: beacon-chain/db/kv/state.go [hunk 2]
```diff
@@ -97,6 +100,39 @@ func (k *Store) SaveState(ctx context.Context, state *pb.BeaconState, blockRoot
...
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.DeleteStates")
+	defer span.End()
+	var wg sync.WaitGroup
+	errs := make([]string, 0)
+	wg.Add(len(blockRoots))
+	for _, r := range blockRoots {
+		go func(w *sync.WaitGroup, root [32]byte) {
+			defer w.Done()
+			if err := k.DeleteState(ctx, root); err != nil {
+				errs = append(errs, err.Error())
```
Comment (shayzluf) [inline beacon-chain/db/kv/state.go]: maybe can add a test for this code path?

Comment (terencechain) [reply beacon-chain/db/kv/state.go]: yea ill add one in the following PR

Comment (terencechain) [reply beacon-chain/db/kv/state.go]: although it's hard to simulate that `err`

Comment (shayzluf) [reply beacon-chain/db/kv/state.go]: you are right `Returns an error if the bucket was created from a read-only transaction`

Comment (shayzluf) [reply beacon-chain/db/kv/state.go]: no need i guess

**Pull request merged**